tests/unit/s2n_connection_serialize_test.c

```diff
@@ -26,6 +26,44 @@
                           0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
                           0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
 
+static S2N_RESULT s2n_test_key_update(struct s2n_connection *client_conn, struct s2n_connection *server_conn)
+{
+    /* One side initiates key update */
+    RESULT_GUARD_POSIX(s2n_connection_request_key_update(client_conn, S2N_KEY_UPDATE_NOT_REQUESTED));
+
+    /* Sending and receiving is successful after key update */
+    EXPECT_OK(s2n_send_and_recv_test(client_conn, server_conn));
+    EXPECT_EQUAL(client_conn->send_key_updated, 1);
+    EXPECT_EQUAL(server_conn->recv_key_updated, 1);
+
+    /* Other side initiates key update */
+    EXPECT_SUCCESS(s2n_connection_request_key_update(server_conn, S2N_KEY_UPDATE_NOT_REQUESTED));
+
+    /* Sending and receiving is successful after key update */
+    EXPECT_OK(s2n_send_and_recv_test(server_conn, client_conn));
+    EXPECT_EQUAL(client_conn->recv_key_updated, 1);
+    EXPECT_EQUAL(server_conn->send_key_updated, 1);
+
+    return S2N_RESULT_OK;
+}
+
+static S2N_RESULT s2n_ticket_pickup(struct s2n_connection *conn, struct s2n_blob *session_ticket)
+{
+    int session_length = s2n_connection_get_session_length(conn);
+    EXPECT_NOT_EQUAL(session_length, 0);
+    EXPECT_SUCCESS(s2n_alloc(session_ticket, session_length));
+    EXPECT_SUCCESS(s2n_connection_get_session(conn, session_ticket->data, session_length));
+    session_ticket->size = session_length;
```

Comment:
I wouldn't expect `session_ticket->size = session_length` to be necessary. Is it?

Might also be a good idea to use realloc here. Not a big deal since this is just a test, but eh

Reply:
s2n_connection_set_session() takes a length parameter, so I do need to put something in there. But yeah, I can do a realloc.